lib/events.js

```diff
@@ -240,23 +243,25 @@ EventEmitter.prototype.removeListener =
 
 EventEmitter.prototype.removeAllListeners =
     function removeAllListeners(type) {
-      var key, listeners;
+      var listeners, events;
 
-      if (!this._events)
+      if (!(events = this._events))
         return this;
 
       // not listening for removeListener, no need to emit
-      if (!this._events.removeListener) {
+      if (!events.removeListener) {
         if (arguments.length === 0)
           this._events = {};
-        else if (this._events[type])
-          delete this._events[type];
+        else if (events[type])
+          delete events[type];
         return this;
       }
 
       // emit removeListener for all listeners on all events
       if (arguments.length === 0) {
-        for (key in this._events) {
+        var keys = Object.keys(events);
```

Comment:
That's not the point, `this._events`  is being treated like a hash table - if it's not normalized by `delete` it will be soon normalized by adding  enough event names on it (16 the last time I checked was the limit on the heuristic). Just forget for-in forever and be happy.